### ISSUE TITLE
How can we expect the model to be ambiguity-aware? 

### ISSUE THREAD

**Issue by liyi01827** [facebookresearch/segment-anything#168]
I am curious why given an ambiguous prompt and by designing 3 mask outputs will make the model ambiguity-aware. 

I do not understand why only optimizing the mask with minimum loss will help this. And how this results in three different mask outputs? I think they can still be similar mask outputs if we are not explicitly making their outputs far from each other.

**Comment by ustcjinggg** [comment]
I believe in many cases, three mask outputs are similar. Though some cases the point can represent many different objects and lead to different mask outputs for one point.

**Comment by franchesoni** [comment]
The way to reduce the training loss is to provide diverse outputs because you take a min. If you didn't provide diverse outputs then you are in the case with a single output, which is ambiguous and as such has higher error than the min of three possible segmentations. Is it clear?

**Comment by liyi01827** [comment]
> The way to reduce the training loss is to provide diverse outputs because you take a min. If you didn't provide diverse outputs then you are in the case with a single output, which is ambiguous and as such has higher error than the min of three possible segmentations. Is it clear?

I agree that this can further reduce the training loss because of the diverse outputs. But it is not my original question. My question is why can you guarantee three diverse outputs? If the three outputs are similar, taking the minimum of them won't help.

I kind of agree with the answer by @ustcjinggg. Most of cases you will have similar three mask outputs but some will give you different outputs.
